src/edit_session.js

```diff
@@ -259,7 +259,32 @@ class EditSession {
         this.setUndoManager(this.$undoManager);
         this.getUndoManager().reset();
     }
-    
+
+    /**
+     * Returns the current edit session as a JSON string.
+     * @private
+     * @method toJSON
+     * @returns {String}
+     */
+    toJSON() {
+        return {
+            annotations: this.$annotations,
+            breakpoints: this.$breakpoints,
+            folds: this.getAllFolds().map(function(fold) {
+                return fold.range;
+            }),
+            history: {
+                undo: this.getUndoManager().$undoStack,
+                redo: this.getUndoManager().$redoStack
+            },
```

Comment:
I think we can combine this with #5236 and make this:

```suggestion
            history: this.getUndoManager().toJSON()
```

Reply:
Yeah that sounds good, but then that PR would need to be merged first I guess.

Reply:
That's correct. Would you be able to merging these with @NotLazy? They mentioned they're going to add tests when they get a chance and then we should be able to merge it.

Reply:
Yeah, we can wait until he adds the tests then you can merge his PR, and I can adjust my PR and we can merge this. Either that, or you can merge this now, then when has added tests and his PR is merged, then you, he or I can file a new PR that changes the `history` property to use his code.

Reply:
@andredcoliveira can you merge this?